File: solutions/674/main.cpp
/* ************************************************************************
> File Name:     main.cpp
> Author:        Sean
> Project:    LeetCode
> Created Time:  Fri Nov 26 11:13:52 2021
> Description:   
 ************************************************************************/
#include <vector>
#include <algorithm> 
#include <gtest/gtest.h> 

using namespace std; 

class Solution {
public:
    int findLengthOfLCIS(vector<int>& nums) {
        // dp table
        // initialize
        vector<int> dp(nums.size(), 1);
        // travese dp table
        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] > nums[i-1]) dp[i] = dp[i-1] + 1;
        }
        return *max_element(dp.begin(), dp.end());
    }
};

TEST(Test674, SimpleTest) {
    vector<int> nums{
        1,3,5,4,7
    };
    Solution s; 
    EXPECT_EQ(s.findLengthOfLCIS(nums), 3);
}

int main() {
    testing::InitGoogleTest();
    return RUN_ALL_TESTS();
}

File: solutions/674/main.py
from typing import List 

class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        # greedy approach
        result = 1

        # check input
        if len(nums) == 1:
            return result

        count = 1

        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                count += 1
            else:
                count = 1

            if result < count:
                result = count

        return result

if __name__ == "__main__": 
    nums = [1,3,5,4,7]
    s = Solution()
    assert s.findLengthOfLCIS(nums) == 3
